### PR TITLE
feat(#750): implement toggle for light/dark mode

### PR DESCRIPTION
**Pull request by Jshen123**
**Description**
Add Theme Toggle
image

**Other changes:**
refactored `useTheme` hook

### CODE REVIEW

File: next/src/components/Menu.tsx [hunk 1]
```diff
@@ -7,14 +7,18 @@ interface MenuProps {
   icon?: ReactNode;
   chevron?: boolean;
   name?: string;
+  className?: string;
```
Comment (asim-shrestha) [inline next/src/components/Menu.tsx:10]: We don't want to manually pass these in anymore

**Comment by asim-shrestha** [comment]
Not available when not logged in

image

File: next/src/styles/globals.css [hunk 2]
```diff
@@ -88,7 +88,7 @@ tr:nth-child(even) {
   .background {
     background-image: radial-gradient(
       circle,
-      hsla(240, 11%, 85%, 0.75) 2px,
+      rgba(58, 58, 58, 0.75) 2px,
```
[the remark this answers is not in the document]
Comment (Jshen123) [reply next/src/styles/globals.css:91]: Hi Asim, this is actually reverting back to rgba. 

In terms of HSL, it's a bit easier for me to work with.

File: next/src/components/Menu.tsx (hunk 1 above)
Comment (Jshen123) [reply next/src/components/Menu.tsx]: ✅

File: next/src/components/Menu.tsx [hunk 3]
```diff
@@ -7,14 +7,17 @@ interface MenuProps {
...
 }
 
-function Menu({ icon, name, items, chevron }: MenuProps) {
+function Menu({ icon, name, items, chevron, buttonPosition = "top" }: MenuProps) {
   return (
     <MenuPrimitive>
       <div className="relative">
-        <MenuPrimitive.Button className="flex h-8 items-center gap-1 rounded-lg border border-white/30 bg-[#3a3a3a] p-2 font-bold hover:border-[#1E88E5]/40 hover:bg-[#6b6b6b]">
+        <MenuPrimitive.Button
+          className={`neutral-button-primary flex h-8 items-center gap-1 rounded-lg  border p-2 font-bold`}
```
[the remark this answers is not in the document]
Comment (Jshen123) [reply next/src/components/Menu.tsx]: ✅

File: next/src/hooks/useTheme.ts [hunk 4]
```diff
@@ -1,36 +1,44 @@
...
-  } else {
+  } else if (!shouldAppThemeBeDark) {
     classList.remove(DARK_THEME);
   }
 };
 
-export const useTheme = (theme: Theme) => {
-  theme = THEMES.includes(theme) ? theme : "system";
+export const useTheme = () => {
+  // theme = THEMES.includes(theme) ? theme : "system";
```
[the remark this answers is not in the document]
Comment (Jshen123) [reply next/src/hooks/useTheme.ts]: ✅

File: next/src/components/Menu.tsx [hunk 5]
```diff
@@ -27,12 +30,16 @@ function Menu({ icon, name, items, chevron }: MenuProps) {
           leaveFrom="transform scale-100 opacity-100"
           leaveTo="transform scale-95 opacity-0"
         >
-          <MenuPrimitive.Items className="absolute right-0 top-full z-20 mt-1 max-h-48 w-fit overflow-hidden rounded-xl border-2 border-white/10 bg-[#3a3a3a] shadow-xl">
+          <MenuPrimitive.Items
+            className={`absolute right-0 ${
+              buttonPosition === "top" ? "top-full mt-1" : "bottom-full mb-9"
+            } background-color-3  z-20 max-h-48 w-fit overflow-hidden rounded-xl border-2 border-white/10 shadow-xl`}
```
[the remark this answers is not in the document]
Comment (Jshen123) [reply next/src/components/Menu.tsx]: ✅

File: next/src/ui/toggleButton.tsx [hunk 6]
```diff
@@ -0,0 +1,24 @@
...
+
+  const icon = props.checked ? props.onIcon : props.offIcon;
+  return (
+    <button type="button" className={props.className} onClick={handleClick}>
+      {icon}
+    </button>
+  );
+};
+
+export default ToggleButton;
```
[the remark this answers is not in the document]
Comment (Jshen123) [reply next/src/ui/toggleButton.tsx]: ✅

**Pull request merged**